main/src/cgeo/geocaching/maps/CGeoMap.java

```diff
@@ -671,13 +671,7 @@ public boolean onOptionsItemSelected(final MenuItem item) {
             case R.id.menu_store_caches:
                 if (!isLoading()) {
                     final Set<String> geocodesInViewport = getGeocodesForCachesInViewport();
-                    final List<String> geocodes = new ArrayList<>();
-
-                    for (final String geocode : geocodesInViewport) {
-                        if (!DataStore.isOffline(geocode, null)) {
-                            geocodes.add(geocode);
-                        }
-                    }
+                    final List<String> geocodes = new ArrayList<>(geocodesInViewport);
```

Comment:
Do we need this copy to prevent concurrent modifications on `geocodesInViewport`? Otherwise, we can simply reuse `geocodesInViewport` by adapting the signature of `storeCaches` (`List<String>` → `Collection<String>`).

Reply:
Can we change the commit message on this pull request to that message or is that set when the pull request is instantiated? 

As for the second question, my guess would be that we do not need that copy, and a redesign of the storeCaches would suffice since it's only use is from this method.

Reply:
If you use `git commit --amend` instead of `git commit` the changes will be added to the current commit and you can change the commit message. Thus, `git commit --amend` without any changes allows you to change the commit message of your current commit.

If you already have multiple commits, you can use interactive rebase to combine them, see https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history.